docs/design-doc-direct-access-links.md

```diff
@@ -0,0 +1,813 @@
+                {
+                    break;
+                }
+            }
+
+            if (directAccessUrl == null)
+            {
+                if (!directAccessUrlSupported)
+                {
+                    // The direct access URL was not supported
+                    throw new UnsupportedOperationException("Retrieving direct access URLs is not supported by this content store.");
+                }
+                else if (!contentUrlSupported)
+                {
+                    // The content URL was not supported
+                    throw new UnsupportedContentUrlException(this, contentUrl);
+                }
+            }
+
+            return directAccessUrl;
+        }
+        finally
+        {
+            readLock.unlock();
+        }
+    }
+```
+
+#### S3 Content Store
+
+![S3 flow - Sequence Diagram](images/dau-design-doc-sequence-diagram-s3-flow-1.svg)
+
+_PlantUML:_
+```puml
+@startuml
+
+participant "DAU Client Thread" as Client
+participant "Agregating Content Store" as ACS
+participant "Caching Content Store" as CCS
+participant "S3 Content Store" as S3CS
+participant "S3 Client Java API\n(AWS SDK)" as Sdk
+
+autonumber
+
+Client -> ACS ++ : Request DAU
+ACS -> CCS ++ : Request DAU
+CCS -> S3CS ++ : Request DAU
+S3CS -> Sdk ++ : Request DAU
+Sdk -> Sdk : Generate DAU
+return DAU
+return DAU
+return DAU
+return DAU
+
+@enduml
+```
+
+#### Multiple Content Stores
+When multiple content stores are configured in ACS, the DAU feature can be enabled/disabled
+individually on each store (provided it supports DAUs).
+
+#### Encrypted stores
+Not supported.
+
+### REST API flow
+The REST API endpoint methods will always call the service layer with the default REST expiry
+time (`alfresco.restApi.directAccessUrl.defaultExpiryTimeInSec`). The service layer will decide
+whether the expiry time must be reduced (for instance when the content store max expiry time is
+lower).
+
+![REST API flow - Sequence Diagram](images/dau-design-doc-sequence-diagram-rest-flow-1.svg)
+
+_PlantUML:_
+```puml
+@startuml
+
+participant "REST Client" as RC
+participant "REST Endpoint\n(ACS Bean)" as EP
+participant "Service Layer\n(ACS ContentService or Nodes/Renditions beans)" as SV
+
+
+alt DAU is disabled (at REST level)
+    RC -> EP ++ : Request DAU
+    EP -> EP : Check if REST DAU is enabled
+    return 501 Unsupported
+|||
+else DAU is enabled at REST level but disabled or unsupported
+|||
+    RC -> EP ++ : Request DAU
+    EP -> EP : Check if REST DAU is enabled
+    EP -> SV ++ : Request DAU
+    return throw DisabledServiceException
+    return 501 Unsupported
+|||
+else DAU is enabled and supported
+    RC -> EP ++ : Request DAU
+    EP -> EP : Check if REST DAU is enabled
+    EP -> SV ++ : Request DAU
+    return DAU
+    return DAU
+end
+
+@enduml
+```
+
+### UI Client Integration example
+For downloading a file from ACS, a client application could implement a logic that first attempts
```

Comment:
I think this needs a detailed design section for ADW itself, ie. exposing this feature end-to-end, including all aspects of the UX, ie. including configuration &/or discovery (from ADW) and whether it will be for Download only (including versions) or also potentially Previews (ie. renditions). 

This should enable a feature team to potentially deliver the complete end-to-end feature, eg.. S3 Connector, Repository APIs and the ADW changes. Ideally also the Azure Connector as well.